src/azure-cli/azure/cli/command_modules/hdinsight/_validators.py

```diff
@@ -145,3 +144,29 @@ def validate_worker(self, cmd, namespace):  # pylint: disable=unused-argument
                         self.resource_name)
         except CLIError as e:
             self.exception = e
+
+
+def validate_timezone_name(namespace):
+    if namespace.timezone:
+        from .util import AUTOSCALE_TIMEZONES
+        zone = next((x for x in AUTOSCALE_TIMEZONES if x.lower() == namespace.timezone.lower()), None)
+        if not zone:
+            raise CLIError(
+                "Invalid time zone: '{}'. Run 'az hdinsight autoscale list-timezones' for values.".format(
+                    namespace.timezone))
+        namespace.timezone = zone
+        return zone
+    return None
+
+
+def validate_time(namespace):
+    if namespace.time:
+        hour, minute = namespace.time.split(':')
```

Comment:
> if user doesn't input `:` at all, this line would cause `ValueError: not enough values to unpack (expected 2, got 1)`. Better to try/catch and refine the error message.

fixed by add extra validation.